src/python/WMComponent/AnalyticsDataCollector/AnalyticsPoller.py

```diff
@@ -77,6 +78,12 @@ def setup(self, parameters):
             pluginFactory = WMFactory("plugins", "WMComponent.AnalyticsDataCollector.Plugins")
             self.plugin = pluginFactory.loadObject(classname=self.pluginName)
 
+        # delete all agent-request type doc.
+        self.localSummaryCouchDB.deleteAllAgentRequestDocument()
+    
+        # populate cache: This will take time ~5 min
+        self.revCache = self.centralWMStatsCouchDB.getAllAgentRequestFromCentralServer()
```

Comment:
How about encapsulating it in a try/except and printing a nice error message if it fails (likely timeout) and assign an empty dict to `self.revCache`?